modules/ServerAPI/src/server/routes/ThingServiceRoutes.coffee

```diff
@@ -285,6 +287,42 @@ updateThing = (thing, testMode, callback) ->
 		)
 
 
+exports.bulkPostThingsSaveFile = (req, resp) ->
+	exports.bulkPostThingsInternal req.body, (response) =>
+		checkFilesAndUpdate = (thing, callback) ->
+			fileVals = serverUtilityFunctions.getFileValuesFromEntity thing, false
+			filesToSave = fileVals.length
+
+			completeThingUpdate = (thingToUpdate)->
+				updateThing thingToUpdate, false, (updatedThing) ->
+					callback updatedThing, 200
+
+			fileSaveCompleted = (passed) ->
+				if !passed
+					callback "file move failed", 500
+				if --filesToSave == 0 then completeThingUpdate(thing)
```

Comment:
What is this `--filesToSave` doing? Decrementing by one?

Reply:
I'm really confused about where the actual changes to the fileValues happen. We seem to just be passing the `thing` that comes back from the "bulk post" and then calling "updateThing" on it. When do the file values actually change?

Reply:
I added comments to clarify but yes, it is decrementing by one.

Reply:
Similar question here... This `fileSaveCompleted` function is only getting the true/false `passed` variable as an input. Where is the value of `thing` coming from?